### PR TITLE
Add `--provider` flag to `system login`

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**

This PR adds provider flag to system login

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
We need a `--provider` flag instead, I believe.

**Comment by leecalcote** [comment]
Meshery supports more than one provider (conceptually any number). For the same use case that someone might want to specify `--none` to script their login, others will also want to supply alternative provider names, e.g. `--provider Meshery`.

Yes, please change it (without necessarily waiting for me to ask twice, pretty please).

**Review by leecalcote — CHANGES_REQUESTED** [review]
Requesting change for holistic consideration of provider flag, not None as a one-off.

**Comment by alphaX86** [comment]
> Meshery supports more than one provider (conceptually any number). For the same use case that someone might want to specify `--none` to script their login, others will also want to supply alternative provider names, e.g. `--provider Meshery`.
> 
> Yes, please change it (without necessarily waiting for me to ask twice, pretty please).

Understood captain, it'll be executed soon (_Have to do this before another comment arises 🌚_)

File: mesheryctl/pkg/utils/auth.go [hunk 1]
```diff
@@ -372,6 +413,26 @@ func selectProviderPrompt(provs map[string]Provider) Provider {
...
+	for _, prov := range provArray {
+		provNames = append(provNames, prov.ProviderName)
+	}
+
+	for i := 0; i < len(provNames); i++ {
+		if provNames[i] == option {
+			return provArray[i], nil
+		}
+	}
+	return provArray[1], fmt.Errorf("the specified provider '%s' is not available. Please try giving correct provider name", option)
```
Comment (piyushsingariya) [inline mesheryctl/pkg/utils/auth.go:433]: why don't we directly iterate the map itself?? and match _lowercased_ keys with the _lowercased_ option

File: mesheryctl/pkg/utils/auth.go [hunk 2]
```diff
@@ -283,6 +283,47 @@ func InitiateLogin(mctlCfg *config.MesheryCtlConfig) ([]byte, error) {
 	return data, nil
 }
 
+// Initiate Login on provider mentioned by user
+func InitiateLoginDirect(mctlCfg *config.MesheryCtlConfig, option string) ([]byte, error) {
```
Comment (piyushsingariya) [inline mesheryctl/pkg/utils/auth.go:287]:
```suggestion
func InitiateDirectLogin(mctlCfg *config.MesheryCtlConfig, option string) ([]byte, error) {
```

**Comment by leecalcote** [comment]
I'm trying to recall the use case... I believe there is the need to use `system login` from scripted environments, in which case, the interactive selector doesn't meet the need.

File: mesheryctl/pkg/utils/auth.go (hunk 1 above)
Comment (alphaX86) [reply mesheryctl/pkg/utils/auth.go:433]: Never thought at that time 😶

**Comment by alphaX86** [comment]
The prompt is messed up when it comes to script execution and even when Unicode is passed, it's messing up totally. That's why the need of `--provider` is needed

File: mesheryctl/pkg/utils/auth.go (hunk 1 above)
Comment (alphaX86) [reply mesheryctl/pkg/utils/auth.go:401]: Function refactored

Comment (piyushsingariya) [reply mesheryctl/pkg/utils/auth.go:401]: I don't think the function is optimised, we're increasing the memory use here, isn't iterating the map itself a good approach in terms of code?

**Comment by alphaX86** [comment]
Almost forgot the map, I always thought its array, I'll change the way soon

File: mesheryctl/pkg/utils/auth.go (hunk 1 above)
Comment (alphaX86) [reply mesheryctl/pkg/utils/auth.go:401]: Will change soon

File: mesheryctl/pkg/utils/auth.go [hunk 3]
```diff
@@ -393,8 +395,8 @@ func chooseProviderDirect(provs map[string]Provider, option string) (Provider, e
 		provNames = append(provNames, prov.ProviderName)
 	}
 
-	for i := 0; i < len(provNames); i++ {
-		if provNames[i] == option {
+	for i := range provNames {
+		if strings.ToLower(provNames[i]) == option {
```
Comment (piyushsingariya) [inline mesheryctl/pkg/utils/auth.go:399]:
```suggestion
		if strings.ToLower(provNames[i]) == strings.ToLower(option) {
```

**Review by leecalcote — APPROVED** [review]

**Pull request merged**